File: Delphi/Ginkgo ADC/Delphi_USB_ADC_Test/ControlADC.pas
unit ControlADC;

interface
const
//Device type definition
VAI_USBADC	=	1;
VCI_USBCAN1	=	3;
VCI_USBCAN2	=	4;
VGI_USBGPIO	=	1;
VII_USBI2C	=	1;
VSI_USBSPI	=	2;
//错误码定义
ERR_SUCCESS					      =	(0);		// 没有错误
ERR_PARAMETER_NULL			  =	(-1);	  // 传入的指针为空指针
ERR_INPUT_DATA_TOO_MUCH		=	(-2);	  // 参数输入个数多余规定个数
ERR_INPUT_DATA_TOO_LESS		=	(-3);	  // 参数输入个数少余规定个数
ERR_INPUT_DATA_ILLEGALITY	=	(-4);	  // 参数传入格式和规定的不符合
ERR_USB_WRITE_DATA			  =	(-5);	  // USB写数据错误
ERR_USB_READ_DATA			    =	(-6);	  // USB读数据错误
ERR_READ_NO_DATA			    =	(-7);	  // 请求读数据时返回没有数据
ERR_OPEN_DEVICE				    =	(-8);	  // 打开设备失败
ERR_CLOSE_DEVICE			    =	(-9);	  // 关闭设备失败
ERR_EXECUTE_CMD				    =	(-10);	// 设备执行命令失败
ERR_SELECT_DEVICE			    =	(-11);	// 选择设备失败
ERR_DEVICE_OPENED			    =	(-12);	// 设备已经打开
ERR_DEVICE_NOTOPEN			  =	(-13);	// 设备没有打开
ERR_BUFFER_OVERFLOW			  =	(-14);	// 缓冲区溢出
ERR_DEVICE_NOTEXIST			  =	(-15);	// 此设备不存在
ERR_LOAD_KERNELDLL			  =	(-16);	// 装载动态库失败
ERR_CMD_FAILED				    =	(-17);	// 执行命令失败错误码
ERR_BUFFER_CREATE			    =	(-18);	// 内存不足
//定义ADC通道
VAI_ADC_CH0		= (1 shl 0);	//ADC_CH0
VAI_ADC_CH1		= (1 shl 1);	//ADC_CH1
VAI_ADC_CH2		= (1 shl 2);	//ADC_CH2
VAI_ADC_CH3		= (1 shl 3);	//ADC_CH3
VAI_ADC_CH4		= (1 shl 4);	//ADC_CH4
VAI_ADC_CH5		= (1 shl 5);	//ADC_CH5
VAI_ADC_CH6		= (1 shl 6);	//ADC_CH6
VAI_ADC_CH7		= (1 shl 7);	//ADC_CH7
VAI_ADC_ALL		= ($FF);    	//ADC_CH_ALL

//函数声明
function VAI_ScanDevice(NeedInit:Byte):Integer; stdcall;
function VAI_OpenDevice(DevType,DevIndex,Reserved:Integer):Integer; stdcall;
function VAI_CloseDevice(DevType,DevIndex:Integer):Integer; stdcall;
function VAI_InitADC(DevType,DevIndex:Integer;Channel:Byte;Period:Word):Integer; stdcall;
function VAI_ReadDatas(DevType,DevIndex:Integer;DataNum:Word;pData:PWord):Integer; stdcall;
implementation

function VAI_ScanDevice;external 'Ginkgo_Driver.dll' name 'VAI_ScanDevice';
function VAI_OpenDevice;external 'Ginkgo_Driver.dll' name 'VAI_OpenDevice';
function VAI_CloseDevice;external 'Ginkgo_Driver.dll' name 'VAI_CloseDevice';
function VAI_InitADC;external 'Ginkgo_Driver.dll' name 'VAI_InitADC';
function VAI_ReadDatas;external 'Ginkgo_Driver.dll' name 'VAI_ReadDatas';
end.

File: Visual Basic.Net/Ginkgo ADC/VB.NET_USB_ADC_Test/USB_ADC_Test/USB_ADC_Test/ControlADC.vb
Option Strict Off
Option Explicit On

Imports System.Runtime.InteropServices
Module ControlADC
    'General Error Code
    Public Const SUCCESS As Int32 = 0                   ' no Error
    Public Const PARAMETER_NULL As Int32 = -1           ' void pointer 
    Public Const INPUT_DATA_TOO_MUCH As Int32 = -2      ' too many parameters
    Public Const INPUT_DATA_TOO_LESS As Int32 = -3      ' too few parameters
    Public Const INPUT_DATA_ILLEGALITY As Int32 = -4    ' illegal parameter
    Public Const USB_WRITE_DATA_ERROR As Int32 = -5     ' USB write data error
    Public Const USB_READ_DATA_ERROR As Int32 = -6      ' USB read data error
    Public Const READ_NO_DATA As Int32 = -7             ' no data return when request data
    Public Const OPEN_DEVICE_FAILD As Int32 = -8        ' failed to open device
    Public Const CLOSE_DEVICE_FAILD As Int32 = -9       ' failed to close device
    Public Const EXECUTE_CMD_FAILD As Int32 = -10       ' the command failed to execute  
    Public Const SELECT_DEVICE_FAILD As Int32 = -11     ' failed to select device
    Public Const DEVICE_OPENED As Int32 = -12           ' device has open
    Public Const DEVICE_NOTOPEN As Int32 = -13          ' device not open
    Public Const BUFFER_OVERFLOW As Int32 = -14         ' buffer overflow
    Public Const DEVICE_NOTEXIST As Int32 = -15         ' device not exist
    Public Const LOAD_KERNELDLL As Int32 = -16          ' failed to load KernelDLL
    Public Const CMD_FAILED As Int32 = -17              ' failed to execute command
    Public Const BUFFER_CREATE As Int32 = -18           ' out of memory
    ' Define Adapter type
    Public Const VAI_USBADC As Int32 = 1
    ' ADC channel
    Public Const VAI_ADC_CH0 As Byte = 1 << 0       'ADC_CH0
    Public Const VAI_ADC_CH1 As Byte = 1 << 1       'ADC_CH1
    Public Const VAI_ADC_CH2 As Byte = 1 << 2       'ADC_CH2
    Public Const VAI_ADC_CH3 As Byte = 1 << 3       'ADC_CH3
    Public Const VAI_ADC_CH4 As Byte = 1 << 4       'ADC_CH4
    Public Const VAI_ADC_CH5 As Byte = 1 << 5       'ADC_CH5
    Public Const VAI_ADC_CH6 As Byte = 1 << 6       'ADC_CH6
    Public Const VAI_ADC_CH7 As Byte = 1 << 7       'ADC_CH7
    Public Const VAI_ADC_ALL As Byte = &HFF         'ADC_ALL
    ' Function
    Declare Function VAI_ScanDevice Lib "Ginkgo_Driver.dll" (ByVal NeedInit As Byte) As Int32
    Declare Function VAI_OpenDevice Lib "Ginkgo_Driver.dll" (ByVal DevType As Int32, ByVal DevIndex As Int32, ByVal Reserved As Int32) As Int32
    Declare Function VAI_CloseDevice Lib "Ginkgo_Driver.dll" (ByVal DevType As Int32, ByVal DevIndex As Int32) As Int32
    ' Initialize ADC
    Declare Function VAI_InitADC Lib "Ginkgo_Driver.dll" (ByVal DevType As Int32, ByVal DevIndex As Int32, ByVal Channel As Byte, ByVal Period As UInt16) As Int32
    ' Get ADC value
    Declare Function VAI_ReadDatas Lib "Ginkgo_Driver.dll" (ByVal DevType As Int32, ByVal DevIndex As Int32, ByVal DataNum As UInt16, ByVal pData() As UInt16) As Int32

End Module
